gmso/lib/potential_templates.py

```diff
@@ -1,158 +1,69 @@
-    def __init__(self,
-                 name='LennardJonesPotential',
-                 expression='4*epsilon*((sigma/r)**12 - (sigma/r)**6)',
-                 independent_variables=None):
-        if independent_variables is None:
-            independent_variables = {'r'}
-        super(PotentialTemplate, self).__init__(
-            name=name,
-            expression=expression,
-            independent_variables=independent_variables,
-            template=True,
-        )
+POTENTIAL_JSONS = list(Path(__file__).parent.glob('jsons/*.json'))
+JSON_DIR = Path.joinpath(Path(__file__).parent, 'jsons')
 
-class MiePotential(PotentialTemplate):
-    def __init__(self,
-                 name='MiePotential',
-                 expression=('(n/(n-m)) * (n/m)**(m/(n-m)) * '
-                            'epsilon * ((sigma/r)**n - (sigma/r)**m)'),
-                 independent_variables=None):
-        if independent_variables is None:
-            independent_variables = {'r'}
-        super(PotentialTemplate, self).__init__(
-            name=name,
-            expression=expression,
-            independent_variables=independent_variables,
-            template=True,
-        )
 
-class BuckinghamPotential(PotentialTemplate):
-    def __init__(self,
-                 name='BuckinghamPotential',
-                 expression='a*exp(-b*r) - c*r**-6',
-                 independent_variables=None):
-        if independent_variables is None:
-            independent_variables = {'r'}
+def _verify_potential_template_keys(_dict, name):
+    assert 'name' in _dict, f"Key name not found in the potential template {name}.json"
+    assert 'expression' in _dict, f"Key expression not found in the potential template {name}.json"
+    assert 'independent_variables' in _dict, f"Key independent_variables not found in the potential template {name}.json"
+    if str(name) != _dict['name']:
+        raise GMSOError(f'Mismatch between Potential name {name} and {_dict["name"]}')
 
-        super(PotentialTemplate, self).__init__(
-            name=name,
-            expression=expression,
-            independent_variables=independent_variables,
-            template=True,
-        )
 
-class HarmonicBondPotential(PotentialTemplate):
-    def __init__(self,
-                 name='HarmonicBondPotential',
-                 expression='0.5 * k * (r-r_eq)**2',
-                 independent_variables={'r'}):
+def _load_template_json(item, json_dir=JSON_DIR):
+    with json_dir.joinpath(f'{item}.json').open('r') as json_file:
+        potential_dict = json.load(json_file)
+        _verify_potential_template_keys(potential_dict, item)
+        return potential_dict
 
-        super(PotentialTemplate, self).__init__(
-            name=name,
-            expression=expression,
-            independent_variables=independent_variables,
-            template=True,
-        )
 
-class HarmonicAnglePotential(PotentialTemplate):
+class PotentialTemplate(Potential):
```

Comment:
Another thing we discussed in the Monday meeting is that this class almost exactly the same as `Potential`, with the exception of the `template=True` argument supplied to the `super.__init__`. Having an entirely separate class just for changing a single flag seems unnecessarily complicated.

Reply:
I lean towards refactoring it in a different PR. I can do it here if that is desirable.